### PR TITLE
156493140 calendar validation

### PR DESCRIPTION
**Pull request by rckopteri**
# Added
 * Added validation to calendar rules. Now user cannot select which week days route is transiting if from/to dates are not set.
 * Added  validation to calendar notification so that missing rule dates are not indicated if they are not missing. Previously user could add few transit days by hand and validator didn't notice that.

### CODE REVIEW

File: ote/src/cljs/ote/app/controller/route/route_wizard.cljs [hunk 1]
```diff
@@ -615,3 +616,20 @@
 
 (defn valid-name [route]
   (if (empty? (get route ::transit/name)) false true))
+
+(defn valid-calendar-rule-dates? [data]
+  (every?
+    (fn [rule]
+      (if (and
```
Comment (tatut) [inline ote/src/cljs/ote/app/controller/route/route_wizard.cljs]: remove `if`,  `and` will return true/false

File: ote/src/cljs/ote/app/controller/route/route_wizard.cljs [hunk 2]
```diff
@@ -615,3 +616,20 @@
...
+            (not (str/blank? (get rule ::transit/from-date)))
+            (not (str/blank? (get rule ::transit/to-date)))
+            )
+        true false))
+    (::transit/service-rules data)))
+
+(defn valid-calendar-from-tro-dates? [data]
+  (if (and
```
Comment (tatut) [inline ote/src/cljs/ote/app/controller/route/route_wizard.cljs]: remove `if`, formatting

File: ote/src/cljs/ote/ui/form_fields.cljs [hunk 3]
```diff
@@ -733,26 +733,29 @@
         [buttons/save {:on-click #(update! (conj (or data []) {}))
                        :label add-label
                        :label-style style-base/button-label-style
-                       :disabled (values/effectively-empty? (last data))}]])]))
+                       :disabled (if add-label-validate
+                                   (not (add-label-validate (last data)))
```
Comment (tatut) [inline ote/src/cljs/ote/ui/form_fields.cljs]: I don't understand the name  `add-label-validate`, it doesn't add a label but determines disabled state

Comment (rckopteri) [reply ote/src/cljs/ote/ui/form_fields.cljs]: add-label should be "add-new-row-button" or something.
Then add-label-validate would be "add-new-row-button-validate". But you wouldn't like that?
I will change it to :add-label-disabled?

**Pull request closed without merging**